test/functional/notes_controller_test.rb

```diff
@@ -95,8 +96,14 @@ def teardown
   test "should get tools" do
     get :tools
 
-    assert_response :success
-    assert_not_nil :notes
+    assert_response :redirect
```

Comment:
This all looks super, really great. I'd like to request an extra assertion here, though -- a `asserts_redirected_to '/methods'` so we're sure where the redirect goes to. Thanks!